Documentation/project-docs/linux-performance-tracing.md

```diff
@@ -81,6 +81,18 @@ Follow these steps to prepare your machine to collect a performance trace.
 
 	The compressed trace file is now stored in the current working directory.
 
+# Resolving Framework Symbols #
+Framework symbols need to be manually generated at the time the trace is collected.  They are different than app-level symbols because the framework is pre-compiled while apps are just-in-time-compiled.
+
+The easiest way to generate framework symbols is to have perfcollect do it for you automatically when the trace is collected.  To do this, crossgen must be present on the collection machine and located next to libcoreclr.so.  This can be done manually using the following steps:
+
+1. Download the CoreCLR nuget package.  For .NET Core 2.0+, go to [https://www.nuget.org/packages/runtime.linux-x64.Microsoft.NETCore.Runtime.CoreCLR/](https://www.nuget.org/packages/runtime.linux-x64.Microsoft.NETCore.Runtime.CoreCLR/).
```

Comment:
I would like the instructions to be a bit more specific.   
After experimenting a bit my recommendation is the following 

Download the runtime.linux-x64.microsoft.netcore.app package for the version of the dotnet runtime you are using.   A simple way of doing this is as the side effect of creating a self-contained version of your application (or frankly any application).   You can do this by going to your application's source directory and typing
```
   dotnet publish --self-contained -r linux-x64
```
This will create a directory bin/*/netcorapp2.0/Linux-x64/publish which contains ALL the files necessary for you app, including the .NET Runtime files and your app exe that you can run)   As a side effect of this it had to download the nuget package for the dotnet runtime (including crossgen).   You should now be able to find it at
```
~/packages/runtime.linux-x64.microsoft.netcore.app/2.0.0/tools/crossgen
```
You need to copy it next to the runtime library (libcoreclr.so) that your app will use.  If you run your application out of your publish directory you just created,  you should copy the crossgen executable there.  If you use the dotnet executable to run your applications DLL, the libcoreclr.so is usually installed /usr/share/dotnet/shared/Microsoft.NETCore.App/* where * is the version number for the version number of the runtime (it should match version number in the path of the crossgen executable).

Once you have copied crossgen next to the libcorclr.so directory the perfCollect script should generate and attach the needed symbolic information for the framework to your *.trace.zip file.    The result is that when you view the trace with PerfView, you will get symbolic names for framework methods.